File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/domain/model/Show.java
package at.ac.tuwien.sepm.groupphase.backend.domain.model;


import lombok.Builder;
import lombok.Value;

import java.time.LocalDateTime;
import java.util.List;

@Value
@Builder(builderClassName = "Builder", toBuilder = true)
public class Show {
    Long showId;
    LocalDateTime createdAt;
    LocalDateTime startTime;
    LocalDateTime endTime;
    Venue venue;
    Event event;
    List<Pricing> pricings;
    SeatingChart seatingChart;
}

File: frontend/src/app/dtos/show.ts
import {Venue} from './venue';
import {Event} from './event';
import {SeatingChart} from './seating-chart';
import {Pricing} from './pricing';
import {Hall} from './hall';

export interface Show {
    showId?: number;
    createdAt?: string;
    startTime: string;
    endTime: string;
    event: Event;
    venue: Venue;
    seatingChart: SeatingChart;
    pricings: Pricing[];
    hall?: Hall;
}
